File: statusneo_backend/src/features/pets/application/create-pet.usecase.ts
import {type Pet } from '../domain/entities/pet.entity';
import {type PetRepository } from '../infrastructure/repositories/pet.repository';
import {type CreatePetInput } from '../presentation/pets.schema';

export class CreatePetUseCase {
  constructor(private readonly petRepository: PetRepository) {}

  async execute(input: CreatePetInput): Promise<Pet> {
    const pet: Pet = {
      id: `pet_${Date.now()}`,
      createdAt: new Date().toISOString(),
      ...input,
    };

    return this.petRepository.create(pet);
  }
}

File: statusneo_assign/lib/features/onboarding/domain/usecases/create_pet_usecase.dart
import '../entities/pet_profile_entity.dart';
import '../repositories/onboarding_repository.dart';

class CreatePetUseCase {
  final OnboardingRepository repository;

  CreatePetUseCase(this.repository);

  Future<String> call(PetProfileEntity pet) {
    return repository.createPet(pet);
  }
}